packages/core/src/editor/NodeHelpers.ts

```diff
@@ -0,0 +1,189 @@
+import { EditorState, Node, NodeId } from "@craftjs/core";
+import invariant from "tiny-invariant";
+import {
+  deprecationWarning,
+  ERROR_CANNOT_DRAG,
+  ERROR_DUPLICATE_NODEID,
+  ERROR_INVALID_NODE_ID,
+  ERROR_MOVE_INCOMING_PARENT,
+  ERROR_MOVE_NONCANVAS_CHILD,
+  ERROR_MOVE_OUTGOING_PARENT,
+  ERROR_MOVE_TO_DESCENDANT,
+  ERROR_MOVE_TO_NONCANVAS_PARENT,
+  ERROR_MOVE_TOP_LEVEL_NODE,
+  ROOT_NODE,
+} from "@craftjs/utils";
+import { serializeNode } from "../utils/serializeNode";
+import { mergeTrees } from "../utils/mergeTrees";
+
+export function NodeHelpers(state: EditorState, id: NodeId) {
+  invariant(typeof id == "string", ERROR_INVALID_NODE_ID);
+
+  const node = state.nodes[id];
+
+  const ref = (id) => NodeHelpers(state, id);
```

Comment:
could we call this `nodeHelpers` or something else? the name `ref` it's a bit confusing since you're calling the method with a different node id.